add fetch_categories and update_preferences to Magicbell::Rails

Mootconn opened the pull request:
Add two new methods for Magicbell::Rails
- fetch_categories: fetches all Magicbell categories
- update_notification_preferences: updates the users notification preferences given a payload
both require external_id

Mootconn committed:
add fetch_categories and update_preferences to Magicbell::Rails

Mootconn committed:
add specs

Mootconn committed:
use VCR instead of stubs

grantspeelman commented on magicbell-rails.gemspec:
```diff
@@ -1,6 +1,6 @@
 Gem::Specification.new do |spec|
   spec.name        = 'magicbell-rails'
-  spec.version     = '0.2.0'
+  spec.version     = '0.2.1'
```
you adding new features so it would be a minor update and not a bug fix update

```suggestion
  spec.version     = '0.3.0'
```

grantspeelman commented on lib/magicbell-rails.rb:
```diff
@@ -11,5 +11,22 @@ module Rails
       Notification.bell(args)
     end
+
+    def self.fetch_categories(external_id)
+      response = client.user_with_external_id(external_id).notification_preferences.retrieve.attributes
+
+      response['categories']&.map { |category| category['slug'] } || []
+    end
```
As new we building a library, we should make it a bit more flexible. So thinking we should return category objects.

then it could be used like this in our code for example:

```ruby
category_slugs = Magicbell::Rails.fetch_categories(external_id).map(&:slug)
```

this way if we ever need any other info of the category in future we could get it.

Also it would be good to keep the top level api methods very simple and push specifics into another class.

```suggestion
    sig { returns(T::Array[UserCategory]) } # we don't have sorbet on this gem but just trying to show you what i think it could return
    def self.fetch_categories(external_id)
      UserCategory.fetch(external_id)
    end
```

grantspeelman commented on lib/magicbell-rails.rb:
```diff
@@ -11,5 +11,22 @@ module Rails
+      response['categories']&.map { |category| category['slug'] } || []
+    end
+
+    def self.update_notification_preferences(external_id, payload)
+      client.user_with_external_id(external_id).notification_preferences.update(payload)
+    end
```
suggestion pushing the logic deeper

```suggestion
    def self.update_notification_preferences(external_id, payload)
      NotificationPerference.update(external_id, payload)
    end
```

grantspeelman reviewed (commented):
Great Work so far 👍🏽

Mootconn committed:
apply code review fixes

- bump version number
- add keyword args to methods
- add NotificationPreference and UserCategory classes.

grantspeelman commented on app/models/magicbell/rails/user_category.rb:
```diff
@@ -0,0 +1,23 @@
+        response = Magicbell::Rails.client.user_with_external_id(external_id)
+                                   .notification_preferences.retrieve.attributes
+
+        response['categories']&.map do |category|
+          UserCategory.new(
+            slug: category['slug'], label: category['label'],
+            channels: category['channels'].map do |channel|
+              Channel.new(slug: channel['slug'], label: channel['label'], enabled: channel['enabled'])
+            end
+          )
+        end || []
```
```suggestion
        Array(response['categories']).map do |category|
          UserCategory.new(
            slug: category['slug'], label: category['label'],
            channels: category['channels'].map do |channel|
              Channel.new(slug: channel['slug'], label: channel['label'], enabled: channel['enabled'])
            end
          )
        end
```

grantspeelman reviewed (approved):
Great Work 👍🏽

Mootconn committed:
add myself as an author

Mootconn committed:
wrap fetch in array
